packages/components/src/components/input/input.ts

```diff
@@ -632,7 +632,7 @@ export default class SdInput extends SolidElement implements SolidFormControl {
                 // eslint-disable-next-line @typescript-eslint/no-explicit-any
                 ifDefined(this.autocomplete as any)
               }
-              autocorrect=${ifDefined(this.type === 'password' ? 'off' : this.autocorrect)}
+              autocorrect=${ifDefined(this.type === 'password' ? 'off' : this.autocorrect ? 'on' : 'off')}
```

Comment:
We need to keep the original types here, otherwise we have a breaking change.

See https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/autocorrect